src/test/java/org/apache/ibatis/submitted/cursor_simple/PostgresCursorTest.java

```diff
@@ -32,31 +30,29 @@
 import org.apache.ibatis.session.SqlSessionFactory;
 import org.apache.ibatis.session.SqlSessionFactoryBuilder;
 import org.apache.ibatis.transaction.jdbc.JdbcTransactionFactory;
-import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
 
-import ru.yandex.qatools.embed.postgresql.EmbeddedPostgres;
-import ru.yandex.qatools.embed.postgresql.util.SocketUtil;
-
-@Tag("EmbeddedPostgresqlTests")
+@Tag("TestcontainersTests")
+@Testcontainers
 class PostgresCursorTest {
 
-  private static final EmbeddedPostgres postgres = new EmbeddedPostgres();
+  @Container
+  private static final PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>().withDatabaseName("cursor_simple")
```

Comment:
Have you thought about using [the JDBC URL-style containers](https://www.testcontainers.org/modules/databases/#database-containers-launched-via-jdbc-url-scheme)?

With it, you don't even need to create any containers and manage their lifecycle with JUnit extensions :)

Reply:
(it works as a meta JDBC driver and creates containers automatically. It will stop the created container when last connection of the requesting driver closes)

Reply:
In our tests, connections are opened and closed in each test, so with the JDBC URL-style, the container restarts between tests which is not ideal.
The JUnit extension's lifecycle, OTOH, fit this test structure.

Is there any other reason to choose JDBC URL-style? like efficiency in resource usage?

Reply:
Ok, I see.

> Is there any other reason to choose JDBC URL-style? like efficiency in resource usage?

Not really :) It uses the same container definition under the hood.

> In our tests, connections are opened and closed in each test, so with the JDBC URL-style, the container restarts between tests which is not ideal.

FYI there is also [a "daemon" mode](https://www.testcontainers.org/modules/databases/#running-container-in-daemon-mode), it will keep the container running even when there're no open connections.
It is useful when you need a single DB instance for all tests, but I see that you're not reusing the container between the tests (although I would suggest to do it to speed things up).
So I will just keep this information here for future reference :)

Reply:
We still don't have many tests involving PostgreSQL, but it might be a good opportunity to consider reusing the container. I'll look into it.
Thanks for the advice!